content/docs/user-guide/hydra-integration.md

```diff
@@ -0,0 +1,251 @@
+# Hydra Integration
+
+_New in DVC 2.25.0 (see `dvc version`)_
+
+In this guide we explain how to enable and use the DVC and
+[Hydra](https://hydra.cc/) integration.
+
+## How it works
+
+On each `dvc exp run` call, DVC will use Hydra to **compose** a single
+configuration object and **dump** it to `params.yaml`. This will happen
+**before** the experiment starts running.
+
+Instead of relying on the
+[`@hydra.main`](https://hydra.cc/docs/tutorials/basic/your_first_app/simple_cli/)
+Python decorator for composing the config object, we can now parse a single
+file: `params.yaml`.
+
+This allows to combine Hydra composition and DVC <abbr>parameters</abbr> in
+order to configure DVC <abbr>pipelines</abbr> (running multiple steps of
+different shell commands, instead of a single Python script) and use features
+like [Templating](/doc/user-guide/project-structure/dvcyaml-files#templating)
+and [`foreach`](/doc/user-guide/project-structure/dvcyaml-files#foreach).
+
+## Setting up Hydra
```

Comment:
Is there an example repo? It's not necessary but seems like it would be very useful here.

Reply:
I have a WIP example repo but need to be polished and didn't want to block the docs. I was hoping that the snippets could be understood for themselves